### PR TITLE
fix(ivy) ensure ngcc can compile `@angular/core`

### PR DESCRIPTION
**Pull request by petebacondarwin**
This is built on top of #25905 - so you can ignore the common commits. This PR is the last three commits, right now.

### CODE REVIEW

File: packages/compiler-cli/src/ngcc/src/host/fesm2015_host.ts [hunk 1]
```diff
@@ -816,6 +833,70 @@ export class Fesm2015ReflectionHost extends TypeScriptReflectionHost implements
...
+      from: importDeclaration.moduleSpecifier.text,
+      name: id.text,
+    };
+  }
+
+  /**
+   * Test whether a decorator was imported from `@angular/core`.
+   *
+   * Is the decorator:
+   * * extermally mported from `@angulare/core`?
```
Comment (alxhub) [inline packages/compiler-cli/src/ngcc/src/host/fesm2015_host.ts]: Spelling 🔥

File: packages/compiler-cli/src/ngtsc/transform/src/translator.ts [hunk 2]
```diff
@@ -206,13 +206,18 @@ class ExpressionTranslatorVisitor implements ExpressionVisitor, StatementVisitor
...
+  visitExternalExpr(ast: ExternalExpr, context: Context): ts.PropertyAccessExpression
+      |ts.Identifier {
     if (ast.value.moduleName === null || ast.value.name === null) {
       throw new Error(`Import unknown module or symbol ${ast.value}`);
     }
-    return ts.createPropertyAccess(
-        ts.createIdentifier(this.imports.generateNamedImport(ast.value.moduleName, ast.value.name)),
-        ts.createIdentifier(ast.value.name));
+    const importIdentifier = this.imports.generateNamedImport(ast.value.moduleName, ast.value.name);
+    if (importIdentifier) {
```
Comment (alxhub) [inline packages/compiler-cli/src/ngtsc/transform/src/translator.ts]: Prefer explicit null checks: `importIdentifer === null`.

(it's more readable and it generates much faster JIT code)

**Review by alxhub — APPROVED** [review]

**Review by gkalpak — APPROVED** [review]

File: integration/ngcc/test-local.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+function rm_cache {
+  rm -rf $cache
+}
+rm_cache
+mkdir $cache
+trap rm_cache EXIT
+
+rm -rf dist
+rm -rf node_modules
+yarn install --cache-folder ../$cache
```
[the remark this answers is not in the document]
Comment (petebacondarwin) [reply integration/ngcc/test-local.sh]: 'coz I just copied it from `run-tests.sh` which probably puts the cache in the `integration` folder?

File: integration/ngcc/test-local.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+yarn test || exit 1
```
[the remark this answers is not in the document]
Comment (petebacondarwin) [reply integration/ngcc/test-local.sh]: again, I copied this from `run-tests.sh`

File: integration/ngcc/test-local.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (petebacondarwin) [reply integration/ngcc/test-local.sh]: I'll fix

File: packages/compiler-cli/src/ngcc/src/packages/transformer.ts [hunk 5]
```diff
@@ -24,10 +24,12 @@ import {FileParser} from '../parsing/file_parser';
 import {Esm2015Renderer} from '../rendering/esm2015_renderer';
 import {Esm5Renderer} from '../rendering/esm5_renderer';
 import {FileInfo, Renderer} from '../rendering/renderer';
+
 import {checkMarkerFile, writeMarkerFile} from './build_marker';
 import {EntryPoint, EntryPointFormat} from './entry_point';
 
 
+
```
[the remark this answers is not in the document]
Comment (petebacondarwin) [reply packages/compiler-cli/src/ngcc/src/packages/transformer.ts]: 🤔

**Review by mhevery — APPROVED** [review]

**Pull request closed without merging**